relays/ethereum/src/ethereum_sync.rs

```diff
@@ -111,7 +111,14 @@ impl HeadersSync {
 	}
 
 	/// Select headers that need to be submitted to the Substrate node.
-	pub fn select_headers_to_submit(&self) -> Option<Vec<&QueuedHeader>> {
+	pub fn select_headers_to_submit(&self, stalled: bool) -> Option<Vec<&QueuedHeader>> {
+		// if we operate in backup mode, we only submit headers when sync has stalled
+		if !stalled {
```

Comment:
Maybe merge the two conditions? This way they read more like the comment sentence above.
```suggestion
		if self.params.sub_tx_mode == SubstrateTransactionMode::Backup && !stalled {
```